board/adi/common-sc598-som/sc598-som.c

```diff
@@ -21,6 +22,8 @@ int board_phy_config(struct phy_device *phydev)
 
 int board_init(void)
 {
+	u32 bmode = (readl(0x3108C004) & 0x00000F00) >> 8;
```

Comment:
arch/arm/mach-sc5xx/soc.c provides sc5xx_get_boot_mode which returns a string. Although the string is not usable here, I'd suggest updating soc.c to include a bmode accessor that handles the RCU addresses, masks, andshifts, and also accounts for SPL_FORCE_BMODE in the same manner, to reuse across boards that are interested.